test/server/cards/01_SOR/events/Waylay.spec.js

```diff
@@ -0,0 +1,54 @@
+describe('Waylay', function() {
+    integration(function() {
+        describe('Waylay\'s ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['waylay'],
+                        groundArena: [{ card: 'pyke-sentinel', upgrades: ['entrenched'] }],
+                    },
+                    player2: {
+                        groundArena: ['wampa', 'superlaser-technician'],
+                        spaceArena: [{ card: 'imperial-interceptor', upgrades: ['academy-training'] }]
+                    }
+                });
+            });
+
+            it('can return a friendly or enemy unit to its owner\'s hand', function () {
```

Comment:
Waylay is kind of unique and we haven't tested the `returnToHand()` stuff much here yet so this is one where I'd actually prefer to add a more robust set of tests (apologies for the lack of clarity on that).

If you don't mind, could you add these cases:
- Add a deployed leader and confirm it doesn't appear as targetable for waylay
- A case where we select a friendly unit to return to hand
- Confirm that the waylay-ed unit can be immediately played again

If you think of any additional ones feel free to add them. One exception is that I do eventually want to use Waylay also for testing things like the Snoke effect, or triggers like "when played," but we will do those in a separate PR since that's more for testing the engine stuff than Waylay itself.

Reply:
I was between `returnToHand()` and `returnToHandFromPlay()` and it was a toss up. I only did a shallow dive but my concerns were mostly around if we play this again the state of the object wont maintain damage etc. I will be honest i didnt deep dive into each of the helper methods. So I'm open to using which ever you prefer, and dont mind doinig more extensive tests